packages/form-state-manager/src/utils/use-subscription.ts

```diff
@@ -74,12 +74,37 @@ const useSubscription = ({ name, initialValue, clearOnUnmount, initializeOnMount
    */
   setDetachedState(state);
 
+  const handleError = (isValid: boolean, error: string | undefined = undefined): void => {
+    setState((prev) => ({
+      ...prev,
+      meta: {
+        ...prev.meta,
+        error,
+        valid: isValid,
+        invalid: !isValid,
+        validating: false
+      }
+    }));
+  };
+
   const handleChange = (event: OnChangeEvent) => {
     const sanitizedValue = sanitizeValue(event);
     change(name, sanitizedValue);
-    // TODO Mutate field state with validation flags
+    // TODO Memoize validation results
```

Comment:
could be opt-in.. I can imagine that sometimes you want to check the same value (async validation in very dynamic applications)

Reply:
Yes. But I would rather make it opt-out.